storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java

```diff
@@ -133,13 +139,43 @@ public void indirectTypeMappingShouldWork() {
         assertSame(typeMapping, storIOContentResolver.internal().typeMapping(TestItemSubclass.class));
     }
 
-    @SuppressWarnings("unchecked")
+    @Test
+    public void indirectTypeMappingShouldCacheValue() {
+        class TestItem {
+
+        }
+
+        //noinspection unchecked
+        final ContentResolverTypeMapping<TestItem> typeMapping = ContentResolverTypeMapping.<TestItem>builder()
+                .putResolver(mock(PutResolver.class))
+                .getResolver(mock(GetResolver.class))
+                .deleteResolver(mock(DeleteResolver.class))
+                .build();
+
+        final StorIOContentResolver storIOContentResolver = DefaultStorIOContentResolver.builder()
+                .contentResolver(mock(ContentResolver.class))
+                .addTypeMapping(TestItem.class, typeMapping)
+                .build();
+
+        class TestItemSubclass extends TestItem {
+
+        }
+
+        // Indirect type mapping should give same type mapping as for parent class
+        assertSame(typeMapping, storIOContentResolver.internal().typeMapping(TestItemSubclass.class));
+
+        // Next call should be faster (we can not check this exactly)
+        // But test coverage tool will check that we executed cache branch
+        assertSame(typeMapping, storIOContentResolver.internal().typeMapping(TestItemSubclass.class));
```

Comment:
Because Type Mapping has cache and I don't know how to properly write unit test for it, but coverage tool (Jacoco) checks that we didn't execute cache branch, so I've added second call to the type mapping and to assert that result will be same and Jacoco will verify that we used cache branch.